core/src/error.rs

```diff
@@ -27,202 +27,205 @@
+    /// The background task has been terminated.
+    #[error("The background task been terminated because: {0}; restart required")]
+    RestartNeeded(String),
+    /// Failed to parse the data.
+    #[error("Parse error: {0}")]
+    ParseError(#[from] serde_json::Error),
+    /// Invalid subscription ID.
+    #[error("Invalid subscription ID")]
+    InvalidSubscriptionId,
+    /// Invalid request ID.
+    #[error("Invalid request ID")]
+    InvalidRequestId,
+    /// Client received a notification with an unregistered method
+    #[error("Unregistered notification method")]
+    UnregisteredNotification(String),
+    /// A request with the same request ID has already been registered.
+    #[error("A request with the same request ID has already been registered")]
+    DuplicateRequestId,
+    /// Method was already registered.
+    #[error("Method: {0} was already registered")]
+    MethodAlreadyRegistered(String),
+    /// Method with that name has not yet been registered.
+    #[error("Method: {0} has not yet been registered")]
+    MethodNotFound(String),
+    /// Subscribe and unsubscribe method names are the same.
+    #[error("Cannot use the same method name for subscribe and unsubscribe, used: {0}")]
+    SubscriptionNameConflict(String),
+    /// Request timeout
+    #[error("Request timeout")]
+    RequestTimeout,
+    /// Configured max number of request slots exceeded.
+    #[error("Configured max number of request slots exceeded")]
+    MaxSlotsExceeded,
+    /// Attempted to stop server that is already stopped.
+    #[error("Attempted to stop server that is already stopped")]
+    AlreadyStopped,
+    /// List passed into `set_allowed_origins` was empty
+    #[error("Must set at least one allowed value for the {0} header")]
+    EmptyAllowList(&'static str),
+    /// Failed to execute a method because a resource was already at capacity
+    #[error("Resource at capacity: {0}")]
+    ResourceAtCapacity(&'static str),
+    /// Failed to register a resource due to a name conflict
+    #[error("Resource name already taken: {0}")]
+    ResourceNameAlreadyTaken(&'static str),
+    /// Failed to initialize resources for a method at startup
+    #[error("Resource name `{0}` not found for method `{1}`")]
+    ResourceNameNotFoundForMethod(&'static str, &'static str),
+    /// Trying to claim resources for a method execution, but the method resources have not been initialized
+    #[error("Method `{0}` has uninitialized resources")]
+    UninitializedMethod(Box<str>),
+    /// Failed to register a resource due to a maximum number of resources already registered
+    #[error("Maximum number of resources reached")]
+    MaxResourcesReached,
+    /// Custom error.
+    #[error("Custom error: {0}")]
+    Custom(String),
+    /// Not implemented for HTTP clients.
+    #[error("Not implemented")]
+    HttpNotImplemented,
+    /// You provided invalid HTTP headers
+    #[error("Invalid http headers provided")]
+    InvalidHeaders,
```

Comment:
```suggestion
    #[error("Invalid http header provided")]
    InvalidHeader,
```

(just because this error is emitted when one attempt to set a single header)